src/xmipp/legacy/libraries/tomo/tomo_tiltseries_detect_misalignment_corr.cpp

```diff
@@ -0,0 +1,573 @@
+void ProgTomoTSDetectMisalignmentCorr::cosineStretching(MultidimArray<double> &ti, double ti_angle_high, double ti_angle_low)
+{
+	Matrix2D<double> sm = getCosineStretchingMatrix(ti_angle_high, ti_angle_low);
+	MultidimArray<double> ti_tmp;
+	ti_tmp = ti;
+
+    applyGeometry(1, ti, ti_tmp, sm, xmipp_transformation::IS_NOT_INV, xmipp_transformation::DONT_WRAP);
+}
+
+
+Matrix2D<double> ProgTomoTSDetectMisalignmentCorr::getCosineStretchingMatrix(double ti_angle_high, double ti_angle_low)
+{
+	double cosFactor = cos(ti_angle_low * PI/180.0) / cos(ti_angle_high * PI/180.0);
+	Matrix2D<double> m(3,3);
+
+	MAT_ELEM(m, 0, 0) = cosFactor;
+	// MAT_ELEM(m, 0, 1) = 0;
+	// MAT_ELEM(m, 0, 2) = 0;
+	// MAT_ELEM(m, 1, 0) = 0;
+	MAT_ELEM(m, 1, 1) = 1;
+	// MAT_ELEM(m, 1, 2) = 0;
+	// MAT_ELEM(m, 2, 0) = 0;
+	// MAT_ELEM(m, 2, 1) = 0;
+	MAT_ELEM(m, 2, 2) = 1;
+
+	return m;
+}
+
+
+Matrix2D<double> ProgTomoTSDetectMisalignmentCorr::maxCorrelationShift(MultidimArray<double> &ti1, MultidimArray<double> &ti2)
+{
+	double shiftX;
+	double shiftY;
+	CorrelationAux aux;
+
+	// bestShift(ti1, ti2, shiftX, shiftY, aux, nullptr, 100);
+
+	// Matrix2D<double> relShift(2, 1);
+
+	// MAT_ELEM(relShift, 0, 0) = shiftX;
+	// MAT_ELEM(relShift, 0, 1) = shiftY;
+
+	// 		// double bestShift(const MultidimArray<double> &I1, const MultidimArray<double> &I2,
+	// 		//            double &shiftX, double &shiftY, CorrelationAux &aux,
+	// 		//            const MultidimArray<int> *mask, int maxShift)
+
+	Matrix2D<double> relShift(2, 1);
+	MultidimArray<double> tsCorr;
+
+	// Shift the particle respect to its symmetric to look for the maximum correlation displacement
+	correlation_matrix(ti1, ti2, tsCorr, aux, true);
+
+	#ifdef DEBUG_OUTPUT_FILES
+	Image<double> saveImage;
+
+	saveImage() = tsCorr;
+	saveImage.write("./tsCorr.mrc");
+	#endif
+
+	auto maximumCorrelation = MINDOUBLE;
+	int xDisplacement = 0;
+	int yDisplacement = 0;
+
+	FOR_ALL_DIRECT_ELEMENTS_IN_ARRAY2D(tsCorr)
+	{
+		double value = DIRECT_A2D_ELEM(tsCorr, i, j);
+
+		if (value > maximumCorrelation)
+		{
+			#ifdef DEBUG_TI_CORR
+			std::cout << "new maximumCorrelation " << value << " at (" << i << ", " << j << ")" << std::endl;
+			#endif
+
+			maximumCorrelation = value;
+			xDisplacement = j;
+			yDisplacement = i;
+		}
+	}
+	
+	MAT_ELEM(relShift, 0, 0) = (double)(xDisplacement) - (double)xSize / 2;
+	MAT_ELEM(relShift, 0, 1) = (double)(yDisplacement) - (double)ySize / 2;
+
+
+	#ifdef DEBUG_TI_CORR
+	std::cout << "maximumCorrelation " << maximumCorrelation << std::endl;
+	std::cout << "xDisplacement " << -(double)(xDisplacement) + (double)xSize / 2 << std::endl;
+	std::cout << "yDisplacement " << -(double)(yDisplacement) + (double)ySize / 2 << std::endl;
+
+	std::cout << "Correlation volume dimensions (" << XSIZE(tsCorr) << ", " << YSIZE(tsCorr) << ")" << std::endl;
+	#endif
+
+	return relShift;
+}
+
+
+// void ProgTomoTSDetectMisalignmentCorr::removeOutliers(MultidimArray<double> &ti)
```

Comment:
Do you want to keep this commented code?

Reply:
Since it is legacy and this function might be useful for future developments I would keep it.